Failure looks related:

```
2020-06-21T23:04:34.6679788Z [INFO] Running io.quarkus.it.smallrye.graphql.GreetingsResourceTestIT
2020-06-21T23:04:34.6707340Z 2020-06-21 23:04:34,625 ERROR [io.sma.graphql] (vert.x-worker-thread-1) SRGQL012000: Data Fetching Error: java.lang.RuntimeException: Can not load class [java.util.List]
2020-06-21T23:04:34.6743698Z 	at io.quarkus.smallrye.graphql.runtime.spi.QuarkusClassloadingService.loadClass(QuarkusClassloadingService.java:47)
2020-06-21T23:04:34.6840062Z 	at io.smallrye.graphql.execution.datafetcher.CollectionCreator.newCollection(CollectionCreator.java:27)
2020-06-21T23:04:34.6845795Z 	at io.smallrye.graphql.execution.datafetcher.helper.AbstractHelper.recursiveTransformCollection(AbstractHelper.java:212)
2020-06-21T23:04:34.6848503Z 	at io.smallrye.graphql.execution.datafetcher.helper.AbstractHelper.recursiveTransform(AbstractHelper.java:74)
2020-06-21T23:04:34.6857122Z 	at io.smallrye.graphql.execution.datafetcher.helper.FieldHelper.transformResponse(FieldHelper.java:30)
2020-06-21T23:04:34.6868951Z 	at io.smallrye.graphql.execution.datafetcher.PropertyDataFetcher.get(PropertyDataFetcher.java:29)
2020-06-21T23:04:34.6883708Z 	at graphql.execution.instrumentation.dataloader.DataLoaderDispatcherInstrumentation.lambda$instrumentDataFetcher$0(DataLoaderDispatcherInstrumentation.java:86)
2020-06-21T23:04:34.6912095Z 	at graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:277)
2020-06-21T23:04:34.6918558Z 	at graphql.execution.ExecutionStrategy.resolveFieldWithInfo(ExecutionStrategy.java:202)
2020-06-21T23:04:34.6981164Z 	at graphql.execution.AsyncExecutionStrategy.execute(AsyncExecutionStrategy.java:74)
2020-06-21T23:04:34.6987113Z 	at graphql.execution.ExecutionStrategy.completeValueForObject(ExecutionStrategy.java:659)
2020-06-21T23:04:34.7011994Z 	at graphql.execution.ExecutionStrategy.completeValue(ExecutionStrategy.java:443)
2020-06-21T23:04:34.7022734Z 	at graphql.execution.ExecutionStrategy.completeField(ExecutionStrategy.java:393)
2020-06-21T23:04:34.7028768Z 	at graphql.execution.ExecutionStrategy.lambda$resolveFieldWithInfo$0(ExecutionStrategy.java:204)
2020-06-21T23:04:34.7034790Z 	at java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:680)
2020-06-21T23:04:34.7040018Z 	at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:658)
2020-06-21T23:04:34.7048680Z 	at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2094)
2020-06-21T23:04:34.7054292Z 	at graphql.execution.ExecutionStrategy.resolveFieldWithInfo(ExecutionStrategy.java:203)
2020-06-21T23:04:34.7059269Z 	at graphql.execution.ExecutionStrategy.resolveField(ExecutionStrategy.java:175)
2020-06-21T23:04:34.7068734Z 	at graphql.execution.AsyncSerialExecutionStrategy.lambda$execute$1(AsyncSerialExecutionStrategy.java:41)
2020-06-21T23:04:34.7075619Z 	at graphql.execution.Async.eachSequentiallyImpl(Async.java:77)
2020-06-21T23:04:34.7092022Z 	at graphql.execution.Async.eachSequentially(Async.java:66)
2020-06-21T23:04:34.7118925Z 	at graphql.execution.AsyncSerialExecutionStrategy.execute(AsyncSerialExecutionStrategy.java:36)
2020-06-21T23:04:34.7127351Z 	at graphql.execution.Execution.executeOperation(Execution.java:167)
2020-06-21T23:04:34.7137795Z 	at graphql.execution.Execution.execute(Execution.java:108)
2020-06-21T23:04:34.7138807Z 	at graphql.GraphQL.execute(GraphQL.java:598)
2020-06-21T23:04:34.7146431Z 	at graphql.GraphQL.parseValidateAndExecute(GraphQL.java:529)
2020-06-21T23:04:34.7165988Z 	at graphql.GraphQL.executeAsync(GraphQL.java:493)
2020-06-21T23:04:34.7174940Z 	at graphql.GraphQL.execute(GraphQL.java:426)
2020-06-21T23:04:34.7180386Z 	at io.smallrye.graphql.execution.ExecutionService.execute(ExecutionService.java:124)
2020-06-21T23:04:34.7186901Z 	at io.smallrye.graphql.execution.ExecutionService.execute(ExecutionService.java:96)
2020-06-21T23:04:34.7206844Z 	at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.doRequest(SmallRyeGraphQLExecutionHandler.java:158)
2020-06-21T23:04:34.7217078Z 	at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.doRequest(SmallRyeGraphQLExecutionHandler.java:151)
2020-06-21T23:04:34.7237764Z 	at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.handlePost(SmallRyeGraphQLExecutionHandler.java:94)
2020-06-21T23:04:34.7275367Z 	at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.doHandle(SmallRyeGraphQLExecutionHandler.java:75)
2020-06-21T23:04:34.7288070Z 	at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.handle(SmallRyeGraphQLExecutionHandler.java:53)
2020-06-21T23:04:34.7297130Z 	at io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLExecutionHandler.handle(SmallRyeGraphQLExecutionHandler.java:32)
2020-06-21T23:04:34.7297689Z 	at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
2020-06-21T23:04:34.7302780Z 	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
2020-06-21T23:04:34.7312291Z 	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
2020-06-21T23:04:34.7324065Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2020-06-21T23:04:34.7332090Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2020-06-21T23:04:34.7340431Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2020-06-21T23:04:34.7353834Z 	at java.lang.Thread.run(Thread.java:834)
2020-06-21T23:04:34.7364997Z 	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:497)
2020-06-21T23:04:34.7432869Z 	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
2020-06-21T23:04:34.7441505Z Caused by: java.security.PrivilegedActionException: java.lang.ClassNotFoundException: java.util.List
2020-06-21T23:04:34.7447590Z 	at java.security.AccessController.doPrivileged(AccessController.java:119)
2020-06-21T23:04:34.7472028Z 	at io.quarkus.smallrye.graphql.runtime.spi.QuarkusClassloadingService.loadClass(QuarkusClassloadingService.java:41)
2020-06-21T23:04:34.7472642Z 	... 45 more
2020-06-21T23:04:34.7476285Z Caused by: java.lang.ClassNotFoundException: java.util.List
2020-06-21T23:04:34.7486886Z 	at com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:60)
2020-06-21T23:04:34.7490868Z 	at java.lang.Class.forName(DynamicHub.java:1197)
2020-06-21T23:04:34.7500016Z 	at io.smallrye.graphql.spi.ClassloadingService.loadClass(ClassloadingService.java:59)
2020-06-21T23:04:34.7510904Z 	at io.quarkus.smallrye.graphql.runtime.spi.QuarkusClassloadingService.lambda$loadClass$0(QuarkusClassloadingService.java:43)
2020-06-21T23:04:34.7518722Z 	at java.security.AccessController.doPrivileged(AccessController.java:117)
2020-06-21T23:04:34.7523320Z 	... 46 more
2020-06-21T23:04:34.7523447Z 
2020-06-21T23:04:34.7701161Z [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.087 s <<< FAILURE! - in io.quarkus.it.smallrye.graphql.GreetingsResourceTestIT
2020-06-21T23:04:34.7713515Z [ERROR] test  Time elapsed: 0.087 s  <<< FAILURE!
2020-06-21T23:04:34.7721631Z java.lang.AssertionError: 
2020-06-21T23:04:34.7721973Z 1 expectation failed.
2020-06-21T23:04:34.7728846Z Response body doesn't match expectation.
2020-06-21T23:04:34.7734639Z Expected: a string containing "Goeie more"
2020-06-21T23:04:34.7772566Z   Actual: {"errors":[{"message":"Server Error","locations":[{"line":23,"column":5}],"path":["load","hellos"],"extensions":{"exception":"java.lang.RuntimeException","classification":"DataFetchingException","code":"runtime"}}],"data":{"load":{"language":"afr","hellos":null}}}
```

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -36,7 +36,7 @@
         <smallrye-health.version>2.2.1</smallrye-health.version>
         <smallrye-metrics.version>2.4.2</smallrye-metrics.version>
         <smallrye-open-api.version>2.0.2</smallrye-open-api.version>
-        <smallrye-graphql.version>1.0.3</smallrye-graphql.version>
+        <smallrye-graphql.version>1.0.4</smallrye-graphql.version>
         <smallrye-opentracing.version>1.3.4</smallrye-opentracing.version>
         <smallrye-fault-tolerance.version>4.3.0</smallrye-fault-tolerance.version>
         <smallrye-jwt.version>2.1.2</smallrye-jwt.version>

--- a/extensions/smallrye-graphql/deployment/pom.xml
+++ b/extensions/smallrye-graphql/deployment/pom.xml
@@ -80,6 +80,12 @@
             <artifactId>quarkus-elytron-security-properties-file</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.opentracing</groupId>
+            <artifactId>opentracing-mock</artifactId>
+            <scope>test</scope>
+        </dependency>
+        
     </dependencies>
 
     <build>

--- a/extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java
+++ b/extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java
@@ -264,6 +264,7 @@ public class SmallRyeGraphQLProcessor {
         classes.add(graphql.schema.GraphQLScalarType.class);
         classes.add(graphql.schema.GraphQLSchema.class);
         classes.add(graphql.schema.GraphQLTypeReference.class);
+        classes.add(List.class);
         return classes.toArray(new Class[] {});
     }
 

--- a/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/GraphQLTest.java
+++ b/extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/GraphQLTest.java
@@ -51,11 +51,11 @@ public class GraphQLTest extends AbstractGraphQLTest {
 
     @Test
     public void testBasicGet() {
-        String pingRequest = getPayload("{\n" +
+        String pingRequest = "{\n" +
                 "  ping {\n" +
                 "    message\n" +
                 "  }\n" +
-                "}");
+                "}";
 
         RestAssured.given().when()
                 .accept(MEDIATYPE_JSON)

--- a/extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java
+++ b/extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java
@@ -2,14 +2,20 @@ package io.quarkus.smallrye.graphql.runtime;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.io.StringReader;
 import java.io.StringWriter;
-import java.nio.charset.StandardCharsets;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
 import java.util.List;
 
 import javax.json.Json;
+import javax.json.JsonBuilderFactory;
 import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
 import javax.json.JsonReader;
+import javax.json.JsonReaderFactory;
 import javax.json.JsonWriter;
+import javax.json.JsonWriterFactory;
 
 import io.quarkus.arc.Arc;
 import io.quarkus.arc.ManagedContext;
@@ -29,12 +35,16 @@ import io.vertx.ext.web.RoutingContext;
 public class SmallRyeGraphQLExecutionHandler implements Handler<RoutingContext> {
     private static boolean allowGet = false;
     private static final String QUERY = "query";
+    private static final String VARIABLES = "variables";
     private static final String OK = "OK";
     private volatile ExecutionService executionService;
     private final CurrentIdentityAssociation currentIdentityAssociation;
+    private static final JsonBuilderFactory jsonObjectFactory = Json.createBuilderFactory(null);
+    private static final JsonReaderFactory jsonReaderFactory = Json.createReaderFactory(null);
+    private static final JsonWriterFactory jsonWriterFactory = Json.createWriterFactory(null);
 
     public SmallRyeGraphQLExecutionHandler(boolean allowGet, CurrentIdentityAssociation currentIdentityAssociation) {
-        this.allowGet = allowGet;
+        SmallRyeGraphQLExecutionHandler.allowGet = allowGet;
         this.currentIdentityAssociation = currentIdentityAssociation;
     }
 
@@ -96,19 +106,40 @@ public class SmallRyeGraphQLExecutionHandler implements Handler<RoutingContext> 
 
     private void handleGet(HttpServerResponse response, RoutingContext ctx) {
         if (allowGet) {
-            List<String> queries = ctx.queryParam(QUERY);
-            if (queries != null && !queries.isEmpty()) {
-                String graphqlGetRequest = queries.get(0);
-                String getResponse = doRequest(graphqlGetRequest.getBytes(StandardCharsets.UTF_8));
-                response.setStatusCode(200)
-                        .setStatusMessage(OK)
-                        .end(Buffer.buffer(getResponse));
+            String query = getQueryParameter(ctx, QUERY);
+            if (query != null && !query.isEmpty()) {
+                try {
+                    String variables = getQueryParameter(ctx, VARIABLES);
+
+                    JsonObjectBuilder input = jsonObjectFactory.createObjectBuilder();
+                    input.add(QUERY, URLDecoder.decode(query, "UTF8"));
+                    if (variables != null && !variables.isEmpty()) {
+                        JsonObject jsonObject = toJsonObject(URLDecoder.decode(variables, "UTF8"));
+                        input.add(VARIABLES, jsonObject);
+                    }
+
+                    String getResponse = doRequest(input.build());
+
+                    response.setStatusCode(200)
+                            .setStatusMessage(OK)
+                            .end(Buffer.buffer(getResponse));
+                } catch (UnsupportedEncodingException ex) {
+                    throw new RuntimeException(ex);
+                }
             } else {
                 response.setStatusCode(204).end();
             }
         } else {
             response.setStatusCode(405).end();
         }
+    }
+
+    private String getQueryParameter(RoutingContext ctx, String parameterName) {
+        List<String> all = ctx.queryParam(parameterName);
+        if (all != null && !all.isEmpty()) {
+            return all.get(0);
+        }
+        return null;
     }
 
     private String getAllowedMethods() {
@@ -121,20 +152,36 @@ public class SmallRyeGraphQLExecutionHandler implements Handler<RoutingContext> 
 
     private String doRequest(final byte[] body) {
         try (ByteArrayInputStream input = new ByteArrayInputStream(body);
-                final JsonReader jsonReader = Json.createReader(input)) {
+                final JsonReader jsonReader = jsonReaderFactory.createReader(input)) {
             JsonObject jsonInput = jsonReader.readObject();
-            JsonObject outputJson = getExecutionService().execute(jsonInput);
-            if (outputJson != null) {
-                try (StringWriter output = new StringWriter();
-                        final JsonWriter jsonWriter = Json.createWriter(output)) {
-                    jsonWriter.writeObject(outputJson);
-                    output.flush();
-                    return output.toString();
-                }
-            }
-            return null;
+            return doRequest(jsonInput);
         } catch (IOException ex) {
             throw new RuntimeException(ex);
+        }
+    }
+
+    private String doRequest(JsonObject jsonInput) {
+        JsonObject outputJson = getExecutionService().execute(jsonInput);
+        if (outputJson != null) {
+            try (StringWriter output = new StringWriter();
+                    final JsonWriter jsonWriter = jsonWriterFactory.createWriter(output)) {
+                jsonWriter.writeObject(outputJson);
+                output.flush();
+                return output.toString();
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+        return null;
+    }
+
+    private static JsonObject toJsonObject(String jsonString) {
+        if (jsonString == null || jsonString.isEmpty()) {
+            return null;
+        }
+
+        try (JsonReader jsonReader = jsonReaderFactory.createReader(new StringReader(jsonString))) {
+            return jsonReader.readObject();
         }
     }
 

--- a/tcks/microprofile-graphql/pom.xml
+++ b/tcks/microprofile-graphql/pom.xml
@@ -22,6 +22,9 @@
                     <dependenciesToScan>
                         <dependency>org.eclipse.microprofile.graphql:microprofile-graphql-tck</dependency>
                     </dependenciesToScan>
+                    <suiteXmlFiles>
+                        <suiteXmlFile>testng.xml</suiteXmlFile>
+                    </suiteXmlFiles>
                 </configuration>
             </plugin>
         </plugins>

--- a/tcks/microprofile-graphql/src/main/java/io/quarkus/tck/graphql/TestInterceptor.java
+++ b/tcks/microprofile-graphql/src/main/java/io/quarkus/tck/graphql/TestInterceptor.java
@@ -1,0 +1,240 @@
+package io.quarkus.tck.graphql;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.DirectoryStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.eclipse.microprofile.graphql.tck.dynamic.ExecutionDynamicTest;
+import org.eclipse.microprofile.graphql.tck.dynamic.execution.TestData;
+import org.jboss.logging.Logger;
+import org.testng.ITestResult;
+import org.testng.TestListenerAdapter;
+
+/**
+ * This allows us to override the input and output of the spec's TCK.
+ * Sometimes this is needed when the spec should be updated but can not yet
+ * 
+ */
+public class TestInterceptor extends TestListenerAdapter {
+    private static final Logger LOG = Logger.getLogger(TestInterceptor.class.getName());
+
+    private static final String EXECUTION_TEST = ExecutionDynamicTest.class.getName();
+    private static final String TEST_SPECIFICATION = "testSpecification";
+
+    private static final String OVERRIDES = "src/main/resources/overrides";
+
+    private Map<String, TestData> testDataMap = new HashMap<>();
+
+    public TestInterceptor() {
+        try {
+            Path folderPath = Paths.get(OVERRIDES);
+            DirectoryStream<Path> overrides = Files.newDirectoryStream(folderPath);
+            Set<Path> overrideFolders = toListOfPaths(overrides);
+            testDataMap.putAll(toMapOfTestData(overrideFolders));
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+
+    }
+
+    private Set<Path> toListOfPaths(DirectoryStream<Path> directoryStream) {
+        Set<Path> directories = new HashSet<>();
+        for (Path p : directoryStream) {
+            try (Stream<Path> paths = Files.walk(p)) {
+                Set<Path> tree = paths.filter(Files::isDirectory)
+                        .collect(Collectors.toSet());
+                directories.addAll(tree);
+            } catch (IOException ex) {
+                LOG.errorf("Ignoring directory [{0}] - {1}", new Object[] { p.getFileName().toString(), ex.getMessage() });
+            }
+        }
+        return directories;
+    }
+
+    private Map<String, TestData> toMapOfTestData(Set<Path> testFolders) {
+        Map<String, TestData> testDataMap = new HashMap<>();
+        for (Path testFolder : testFolders) {
+            if (!testFolder.getFileName().toString().startsWith("META-INF")) {// Ignore META-INF
+                try {
+                    TestData testData = toTestData(testFolder);
+                    if (!testData.shouldIgnore()) {
+                        testDataMap.put(testData.getName(), testData);
+                    }
+                } catch (IOException ioe) {
+                    LOG.errorf("Could not add test case {0} - {1}",
+                            new Object[] { testFolder.getFileName().toString(), ioe.getMessage() });
+                }
+            }
+        }
+        return testDataMap;
+    }
+
+    private TestData toTestData(Path folder) throws IOException {
+        TestData testData = new TestData(folder.getFileName().toString().replace("/", ""));
+        Files.walkFileTree(folder, new HashSet<>(), 1, new FileVisitor<Path>() {
+
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                    throws IOException {
+
+                if (!Files.isDirectory(file)) {
+                    String filename = file.getFileName().toString();
+
+                    switch (filename) {
+                        case "input.graphql": {
+                            String content = getFileContent(file);
+                            testData.setInput(content);
+                            break;
+                        }
+                        case "httpHeader.properties": {
+                            Properties properties = new Properties();
+                            properties.load(Files.newInputStream(file));
+                            testData.setHttpHeaders(properties);
+                            break;
+                        }
+                        case "output.json": {
+                            String content = getFileContent(file);
+                            testData.setOutput(content);
+                            break;
+                        }
+                        case "variables.json": {
+                            String content = getFileContent(file);
+                            testData.setVariables(toJsonObject(content));
+                            break;
+                        }
+                        case "test.properties": {
+                            Properties properties = new Properties();
+                            properties.load(Files.newInputStream(file));
+                            testData.setProperties(properties);
+                            break;
+                        }
+                        case "cleanup.graphql": {
+                            String content = getFileContent(file);
+                            testData.setCleanup(content);
+                            break;
+                        }
+                        case "prepare.graphql": {
+                            String content = getFileContent(file);
+                            testData.setPrepare(content);
+                            break;
+                        }
+                        default:
+                            LOG.warnf("Ignoring unknown file {0}", filename);
+                            break;
+                    }
+                }
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFileFailed(Path file, IOException exc)
+                    throws IOException {
+                LOG.errorf("Could not load file {0}[{1}]", new Object[] { file, exc.getMessage() });
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+                return FileVisitResult.CONTINUE;
+            }
+        });
+
+        return testData;
+    }
+
+    private JsonObject toJsonObject(String jsonString) {
+        if (jsonString == null || jsonString.isEmpty()) {
+            return null;
+        }
+        try (JsonReader jsonReader = Json.createReader(new StringReader(jsonString))) {
+            return jsonReader.readObject();
+        }
+    }
+
+    private String getFileContent(Path file) throws IOException {
+        return new String(Files.readAllBytes(file));
+    }
+
+    @Override
+    public void onTestStart(ITestResult itr) {
+
+        if (itr.getTestClass().getName().equals(EXECUTION_TEST)) {
+            TestData testData = getExecutionTestData(itr.getParameters());
+
+            LOG.info("\n\t =================================================="
+                    + "\n\t Testing [" + testData.getName() + "]"
+                    + "\n\t =================================================="
+                    + "\n");
+
+            if (itr.getName().startsWith(TEST_SPECIFICATION) && !testDataMap.isEmpty()
+                    && testDataMap.containsKey(testData.getName())) {
+
+                TestData override = testDataMap.get(testData.getName());
+
+                if (override.getCleanup() != null && !override.getCleanup().isEmpty()) {
+                    testData.setCleanup(override.getCleanup());
+                    LOG.warn("\n\t NOTE: Overriding Cleanup");
+                }
+                if (override.getHttpHeaders() != null && !override.getHttpHeaders().isEmpty()) {
+                    testData.setHttpHeaders(override.getHttpHeaders());
+                    LOG.warn("\n\t NOTE: Overriding HTTP Headers");
+                }
+                if (override.getInput() != null && !override.getInput().isEmpty()) {
+                    testData.setInput(override.getInput());
+                    LOG.warn("\n\t NOTE: Overriding Input");
+                }
+                if (override.getOutput() != null && !override.getOutput().isEmpty()) {
+                    testData.setOutput(override.getOutput());
+                    LOG.warn("\n\t NOTE: Overriding Output");
+                }
+                if (override.getPrepare() != null && !override.getPrepare().isEmpty()) {
+                    testData.setPrepare(override.getPrepare());
+                    LOG.warn("\n\t NOTE: Overriding Prepare");
+                }
+                if (override.getProperties() != null && !override.getProperties().isEmpty()) {
+                    testData.setProperties(override.getProperties());
+                    LOG.warn("\n\t NOTE: Overriding Properties");
+                }
+                if (override.getVariables() != null && !override.getVariables().isEmpty()) {
+                    testData.setVariables(override.getVariables());
+                    LOG.warn("\n\t NOTE: Overriding Variables");
+                }
+            }
+
+        }
+        super.onTestStart(itr);
+
+    }
+
+    private TestData getExecutionTestData(Object[] parameters) {
+        for (Object param : parameters) {
+            if (TestData.class.isInstance(param)) {
+                return TestData.class.cast(param);
+            }
+        }
+        return null;
+    }
+
+}

--- a/tcks/microprofile-graphql/src/main/resources/overrides/createNewNullNamedTeam/output.json
+++ b/tcks/microprofile-graphql/src/main/resources/overrides/createNewNullNamedTeam/output.json
@@ -1,0 +1,14 @@
+{
+    "errors": [
+        {
+            "message": "The field at path '/team/name' was declared as a non null type, but the code involved in retrieving data has wrongly returned a null value.  The graphql specification requires that the parent field be set to null, or if that is non nullable that it bubble up null to its parent and so on. The non-nullable type is 'String' within parent type 'Team'",
+            "path": [
+                "team",
+                "name"
+            ]
+        }
+    ],
+    "data": {
+        "team": null
+    }
+}

--- a/tcks/microprofile-graphql/testng.xml
+++ b/tcks/microprofile-graphql/testng.xml
@@ -1,0 +1,11 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<suite name="MicroProfile_GraphQL" verbose="1">
+    <listeners>
+        <listener class-name="io.quarkus.tck.graphql.TestInterceptor" />
+    </listeners>
+    <test name="All">
+        <packages>
+            <package name="org.eclipse.microprofile.graphql.*"/>
+        </packages>
+    </test>
+</suite>

